grid/pdf-export-server/PdfExport/RadSpreadProcessingExporter.cs

```diff
@@ -0,0 +1,142 @@
+using Telerik.DataSource;
+using Telerik.DataSource.Extensions;
+using System.Threading.Tasks;
+using System.Reflection;
+using Telerik.Documents.Common.Model;
+
+namespace PdfExport
+{
+    public class RadSpreadProcessingExporter
+    {
+        public async Task<byte[]> ExportPdf<T>(IQueryable<T> data, DataSourceRequest gridRequest)
+        {
+            var workbook = new Workbook();
+            workbook.Name = "workbook-1";
+
+            // performance
+            workbook.History.IsEnabled = false;
+            workbook.SuspendLayoutUpdate();
+
+            workbook.Sheets.Add(SheetType.Worksheet);
+            Worksheet worksheet = workbook.ActiveWorksheet;
+
+            var dataResult = await data.ToDataSourceResultAsync(gridRequest);
+            List<T> dataToExport = (dataResult.Data as IEnumerable<T>).ToList();
+
+            int currRow = 0;
+
+            Type typeParameterType = typeof(T);
+            var fieldsList = typeParameterType.GetProperties();
+
+            //some styling for the cells - borders in this example
+            ThemableColor black = new ThemableColor(Telerik.Documents.Media.Color.FromArgb(0, 0, 0, 0));
+            CellBorders desiredBorders = new CellBorders(
+                    new CellBorder(CellBorderStyle.Medium, black),   // Left border 
+                    new CellBorder(CellBorderStyle.Medium, black),   // Top border 
+                    new CellBorder(CellBorderStyle.Medium, black),   // Right border 
+                    new CellBorder(CellBorderStyle.Medium, black),   // Bottom border 
+                    new CellBorder(CellBorderStyle.Thin, black),       // Inside horizontal border 
+                    new CellBorder(CellBorderStyle.Thin, black),       // Inside vertical border 
+                    new CellBorder(CellBorderStyle.None, black),     // Diagonal up border 
+                    new CellBorder(CellBorderStyle.None, black)    // Diagonal down border 
+                );
+
+            for (int i = 0; i < fieldsList.Length; i++)
+            {
+                CellSelection currCell = worksheet.Cells[0, i];
+                currCell.SetValue(fieldsList[i].Name);
+                currCell.SetBorders(desiredBorders);
+            }
+
+            currRow++;
+
+
+            // content
+            for (int i = 0; i < dataToExport.Count; i++)
+            {
+                for (int columnIndex = 0; columnIndex < fieldsList.Length; columnIndex++)
+                {
+                    var cellValue = GetFieldValue(dataToExport[i], fieldsList[columnIndex].Name);
+                    CellSelection currCell = worksheet.Cells[currRow, columnIndex];
+                    SetCellValue(currCell, cellValue);
+                    currCell.SetBorders(desiredBorders);
+                }
+                currRow++;
+            }
+
+            //column sizes
+            for (int i = 0; i < fieldsList.Length; i++)
+            {
+                worksheet.Columns[i].AutoFitWidth();
```

Comment:
The last time I checked the `AutoFitWidth` method had a noticeable impact on the performance in wasm. The call can be avoided with some hard-coded widths instead.